enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+       reason: IngressControllerProgressing
+       status: "True"
+       type: Progressing
+     - lastTransitionTime: "2024-04-10T00:20:50Z"
+       status: "False"
+       type: Degraded
+     - lastTransitionTime: "2024-04-10T00:20:17Z"
+       message: IngressController is upgradeable.
+       reason: Upgradeable
+       status: "True"
+       type: Upgradeable
+     - lastTransitionTime: "2024-04-10T00:20:17Z"
+       message: No evaluation condition is detected.
+       reason: NoEvaluationCondition
+       status: "False"
+       type: EvaluationConditionsDetected
+   domain: apps1.apps.misalunkeip16.devcluster.openshift.com
+   endpointPublishingStrategy:
+     loadBalancer:
+       dnsManagementPolicy: Managed
+       providerParameters:
+         aws:
+           networkLoadBalancer:
+             eipAllocations:
+               - eipalloc-0956fea34de4cb7ab
+               - eipalloc-0e9a3077a70de050a
+               - eipalloc-0b69fc4691f54cdd0
+               - eipalloc-01e6ba6cbba1a391b
+               - eipalloc-0e242df173f906112
+           type: NLB
+         type: AWS
+       scope: External
+     type: LoadBalancerService
+   observedGeneration: 3
+   selector: ingresscontroller.operator.openshift.io/deployment-ingresscontroller=test
+   tlsProfile:
+     ciphers:
+       - ECDHE-ECDSA-AES128-GCM-SHA256
+       - ECDHE-RSA-AES128-GCM-SHA256
+       - ECDHE-ECDSA-AES256-GCM-SHA384
+       - ECDHE-RSA-AES256-GCM-SHA384
+       - ECDHE-ECDSA-CHACHA20-POLY1305
+       - ECDHE-RSA-CHACHA20-POLY1305
+       - DHE-RSA-AES128-GCM-SHA256
+       - DHE-RSA-AES256-GCM-SHA384
+       - TLS_AES_128_GCM_SHA256
+       - TLS_AES_256_GCM_SHA384
+       - TLS_CHACHA20_POLY1305_SHA256
+     minTLSVersion: VersionTLS12
+
+```
+
+#### Cluster admin wants operator to update the Load Balancer service for updated Ingress Controller eipAllocations 
+
+Administrator can tell the Ingress Operator to complete the updating of Load Balancer service automatically
+by annotating the IngressController with the annotation called `ingress.operator.openshift.io/auto-delete-load-balancer=` 
+and then changing the eipAllocations by using the following commands:
+
+Annotate ingress controller with `ingress.operator.openshift.io/auto-delete-load-balancer=`
+```sh
+oc -n openshift-ingress-operator annotate ingresscontrollers/test ingress.operator.openshift.io/auto-delete-load-balancer=
+```
+
+Update the Ingress Controller with the following command:
+
+```sh
+% oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"providerParameters":{"type":"AWS","aws":{"type":"NLB","nlbParameters":{"eipAllocations":[]}}}}}}}'
+```
```

Comment:
We don't need to tell the user to patch if they are already making the update, right?
```suggestion
Update the Ingress Controller with the changes to the EIPs.

```

Reply:
patch command is one way to update the IC which needs to be done after setting the auto-delete-loadbalancer annotation if one wants the IC to recreate the service with eipAllocations

Reply:
so basically to update one can use oc edit or oc patch